src/Serializer/AddResourceClassNormalizer.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace ApiPlatform\Core\Serializer;
+
+use ApiPlatform\Core\Api\IriConverterInterface;
+use ApiPlatform\Core\Bridge\Elasticsearch\Serializer\ItemNormalizer as ElasticsearchItemNormalizer;
+use ApiPlatform\Core\Api\ResourceClassResolverInterface;
+use ApiPlatform\Core\Exception\InvalidArgumentException;
+use ApiPlatform\Core\Util\ClassInfoTrait;
+use Symfony\Component\Serializer\Normalizer\CacheableSupportsMethodInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+
+/**
+ * Add resource class to the context
+ */
+final class AddResourceClassNormalizer implements NormalizerInterface, DenormalizerInterface, CacheableSupportsMethodInterface
+{
+    use ContextTrait;
+    use ClassInfoTrait;
+
+    /** @var DenormalizerInterface|NormalizerInterface */
+    private $normalizer;
+
+    private $resourceClassResolver;
+
+    private $localCache = [];
+
+    private $allowUnmappedClass;
+
+    private $iriConverter;
+
+    public function __construct(NormalizerInterface $normalizer, ResourceClassResolverInterface $resourceClassResolver, IriConverterInterface $iriConverter, bool $allowUnmappedClass = false)
```

Comment:
You can remove `allowUnmappedClass` in a new class. It has been added only for BC in the current class. Here we must always support unmapped classes.